projects/RabbitMQ.Client/ConsumerDispatching/ConsumerDispatcherChannelBase.cs

```diff
@@ -85,11 +85,23 @@ internal ConsumerDispatcherChannelBase(Impl.Channel channel, ushort concurrency)
 
         public ValueTask HandleBasicConsumeOkAsync(IAsyncBasicConsumer consumer, string consumerTag, CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
+
             if (false == _disposed && false == _quiesce)
             {
-                AddConsumer(consumer, consumerTag);
-                WorkStruct work = WorkStruct.CreateConsumeOk(consumer, consumerTag);
-                return _writer.WriteAsync(work, cancellationToken);
+                try
+                {
+                    AddConsumer(consumer, consumerTag);
+                    WorkStruct work = WorkStruct.CreateConsumeOk(consumer, consumerTag);
+
+                    cancellationToken.ThrowIfCancellationRequested();
+                    return _writer.WriteAsync(work, cancellationToken);
```

Comment:
Are you ok with this catch never being called for exceptions raised by `WriteAsync`? Because otherwise I would encourage using proper async / await here which might also allow you to remove the additional `ThrowIfCancellationRequested`

Reply:
Ah! Thank you for noticing!